src/components/text-search/search-as-you-type.js

```diff
@@ -0,0 +1,344 @@
+    // See: https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_pushState()_method
+    if (window.history.pushState) {
+      const newUrl = `${window.location.protocol}//${window.location.host}${window.location.pathname}?${stringified}`;
+      window.history.pushState({ path: newUrl }, '', newUrl);
+    } else {
+      window.location.search = stringified;
+    }
+  }
+
+  // When the input is focused, Autosuggest will consult this function
+  // when to render suggestions. Use it, for example, if you want to
+  // display suggestions when input value is at least 2 characters long.
+  shouldRenderSuggestions(value) {
+    const numChars = this.props.autocomplete.numChars || 2;
+    return value.trim().length > numChars;
+  }
+
+  renderSuggestionsContainer({ containerProps, children, query }) {
+    const { mode } = this.props.autocomplete;
+    const titleText = this.props.autocomplete[mode].titleText || 'What are you looking for?';
+    const directionsText = this.props.autocomplete[mode].directionsText || true;
+    const suggestionsWrapperClasses = directionsText
+      ? 'react-autosuggest__suggestions-itemslist-wrapper react-autosuggest__suggestions-itemslist-wrapper--with-directions'
+      : 'react-autosuggest__suggestions-itemslist-wrapper';
+
+    return (
+      <div {... containerProps}>
+        <div className="react-autosuggest__container-title">
+          {titleText}
+          <button className="react-autosuggest__container-close-button" onClick={this.onSuggestionsClearRequested}>x</button>
+        </div>
+        <div className={suggestionsWrapperClasses}>
+          {children}
+        </div>
+        {/* @todo add logic for suggestion mode and alter directionsText accordingly */}
+        {directionsText &&
+          <div className="react-autosuggest__container-directions">
+            <span className="react-autosuggest__container-directions-item">Press <code>ENTER</code> to search for <strong>{query}</strong> or <code>ESC</code> to close.</span>
+            <span className="react-autosuggest__container-directions-item">Press ↑ and ↓ to highlight a suggestion then <code>ENTER</code> to be redirected to that suggestion.</span>
+          </div>
+        }
+      </div>
+    );
+  }
+
+  /**
+   * Define how suggestions are rendered.
+   * Note: must be a pure function.
+   *
+   * @param suggestion
+   *   The suggestion to render
+   *
+   * @param query
+   * Used to highlight the matching string. As user types in the input,
+   * query will be equal to the trimmed value of the input. Then, if user
+   * interacts using the Up or Down keys, the input will get the value of
+   * the highlighted suggestion, but query will remain to be equal to the
+   * trimmed value of the input prior to the Up and Down interactions.
+   *
+   * unused - isHighlighted
+   * Whether or not the suggestion is highlighted.
+   *
+   * @return a string or a ReactElement
```

Comment:
This variable return is a code smell.

Reply:
Makes sense, thanks.  I suppose we could return a ReactElement in either instance. The text contents inside the element are only transformed (with `<b></b>`) if the passed in string matches the regex.  Would that be better?

Reply:
Updated in https://github.com/palantirnet/federated-search-react/pull/26/commits/f1bfc0ff259b826acd32f3beb09c818e9c9cc3a3